### ISSUE TITLE
Having trouble installing via NPM

### ISSUE THREAD

**Issue by SilentCicero** [iurimatias/EtherSim#1]
nick@nick:~/Github$ sudo npm install -g ethersim
[sudo] password for nick: 
npm ERR! not a package /tmp/npm-16321-owQ1nK8p/github.com/iurimatias/node-jsonrpc2.git
npm ERR! Error: ENOENT, open '/tmp/npm-16321-owQ1nK8p/github.com/iurimatias/node-jsonrpc2.git-unpack/package.json'
npm ERR! If you need help, you may report this _entire_ log,
npm ERR! including the npm and node versions, at:
npm ERR!     http://github.com/npm/npm/issues

npm ERR! System Linux 4.0.0-040000-generic
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install" "-g" "ethersim"
npm ERR! cwd /home/nick/Github
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.4.21
npm ERR! path /tmp/npm-16321-owQ1nK8p/github.com/iurimatias/node-jsonrpc2.git-unpack/package.json
npm ERR! code ENOENT
npm ERR! errno 34
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/nick/Github/npm-debug.log
npm ERR! not ok code 0

having trouble installing ethersim from NPM, any thoughts?

**Comment by iurimatias** [comment]
what version of npm and node do you got? it seems to be some permissions issue e.g https://github.com/npm/npm/issues/3664

**Comment by SilentCicero** [comment]
I'll run an update, if I still get the same errors I'll run it by you.

**Comment by iurimatias** [comment]
@SilentCicero just released a new version with a lots of npm issues fixed. please let me know if you are still having issues

**Comment by taoeffect** [comment]
@iurimatias I figured out why Embark is failing to install on OS X.

EtherSim is using an outdated version of `ethereumjs-vm`. The latest version of `ethereumjs-vm` is `1.0.1`, and it has the latest secp256k1 with the fix for OS X.

Unless there are other uncaught outdated dependencies, addressing this should fix:
- https://github.com/ethereum/ethereumjs-tx/issues/6
- https://github.com/iurimatias/embark-framework/issues/99

EDIT: Since this is different from what @SilentCicero describes, I've opened #2.

**Comment by tcoulter** [comment]
We really need to update NPM with what's on develop. :)

**Comment by iurimatias** [comment]
released 0.2.0 with the latest changes

**Comment by taoeffect** [comment]
Tried to `npm install -g EtherSim`, still fails due to unfixed secp256k1 versions:

secp256k1@1.1.4 install /usr/local/lib/node_modules/ethersim/node_modules/ethereumjs-vm/node_modules/secp256k1

secp256k1@1.1.4 install /usr/local/lib/node_modules/ethersim/node_modules/ethereumjs-block/node_modules/secp256k1

**Comment by AronVanAmmers** [comment]
I'm getting a recent version of `secp256k1` when installing from `npm` today. So this issue might be closed?

End result of `npm install -g ethersim`:

```
ethersim@0.2.0 /usr/lib/node_modules/ethersim
├── async@1.5.2
├── shelljs@0.5.3
├── bignumber.js@2.0.7
├── http-proxy@1.13.1 (eventemitter3@1.1.1, requires-port@1.0.0)
├── body-parser@1.14.2 (bytes@2.2.0, content-type@1.0.1, depd@1.1.0, qs@5.2.0, http-errors@1.3.1, on-finished@2.3.0, raw-body@2.1.5, debug@2.2.0, type-is@1.6.11, iconv-lite@0.4.13)
├── yargs@3.32.0 (camelcase@2.1.0, window-size@0.1.4, y18n@3.2.0, decamelize@1.1.2, os-locale@1.4.0, cliui@3.1.0, string-width@1.0.1)
├── request@2.69.0 (is-typedarray@1.0.0, aws-sign2@0.6.0, forever-agent@0.6.1, caseless@0.11.0, stringstream@0.0.5, tunnel-agent@0.4.2, oauth-sign@0.8.1, form-data@1.0.0-rc3, isstream@0.1.2, json-stringify-safe@5.0.1, extend@3.0.0, node-uuid@1.4.7, qs@6.0.2, combined-stream@1.0.5, aws4@1.2.1, tough-cookie@2.2.1, mime-types@2.1.9, bl@1.0.2, http-signature@1.1.1, hawk@3.1.3, har-validator@2.0.6)
├── jayson@1.2.2 (eyes@0.1.8, commander@1.3.2, JSONStream@1.0.3, lodash@3.6.0)
├── web3@0.8.1 (xmlhttprequest@1.8.0, bignumber.js@2.0.7, crypto-js@3.1.6)
├── ethereumjs-util@2.6.0 (rlp@2.0.0, browserify-sha3@0.0.1, bn.js@4.10.3, secp256k1@2.0.10, sha3@1.2.0)
├── merkle-patricia-tree@1.1.2 (semaphore@1.0.3, rlp@1.1.2, browserify-sha3@0.0.0, readable-stream@2.0.5, level-ws@0.0.0, memdown@1.1.2, levelup@1.3.1, ethereumjs-util@1.5.1)
├── ethereumjs-account@1.0.5 (rlp@2.0.0, ethereumjs-util@2.3.1)
├── ethereumjs-tx@0.6.14 (ethereum-common@0.0.13, secp256k1@2.0.10)
├── ethereumjs-block@1.2.1 (ethereum-common@0.0.16, es6-shim@0.34.4, merkle-patricia-tree@2.1.1, ethereumjs-tx@0.7.2)
└── ethereumjs-vm@1.1.0 (functional-red-black-tree@1.0.1, rlp@2.0.0, es6-shim@0.33.13, ethereum-common@0.0.14, async-eventemitter@0.2.2, merkle-patricia-tree@2.1.1, secp256k1@2.0.10)
```

**Comment by taoeffect** [comment]
Yep, I can confirm that this works now (and installs [an insane](http://www.mikeperham.com/2016/02/09/kill-your-dependencies/) amount of dependencies).
